apache-rat-core/src/main/java/org/apache/rat/analysis/HeaderCheckWorker.java

```diff
@@ -47,98 +53,103 @@ class HeaderCheckWorker {
 
     private final int numberOfRetainedHeaderLines;
     private final BufferedReader reader;
-    private final ILicense license;
+    private final Collection<ILicense> licenses;
     private final Document document;
 
-    private int headerLinesToRead;
-    private boolean finished = false;
+    /**
+     * Read the input and perform the header check.
+     *
+     * The number of lines indicates how many lines from the top of the file will be read for processing
+     * 
+     * @param reader The reader for the document.
+     * @param numberOfLines the number of lines to read from the header.
+     * @return The IHeaders instance for the header.  
+     * @throws IOException on input failure
+     */
+    public static IHeaders readHeader(BufferedReader reader, int numberOfLines) throws IOException {
+        final StringBuilder headers = new StringBuilder();
+        int headerLinesRead = 0;
+        String line;
+
+        while (headerLinesRead < numberOfLines && (line = reader.readLine()) != null) {
+            headers.append(line).append("\n");
+        }
+        final String raw = headers.toString();
+        final String pruned = FullTextMatcher.prune(raw).toLowerCase(Locale.ENGLISH);
+        return new IHeaders() {
+            @Override
+            public String raw() {
+                return raw;
+            }
+
+            @Override
+            public String pruned() {
+                return pruned;
+            }
+
+            @Override
+            public String toString() {
+                return "HeaderCheckWorker";
```

Comment:
Should we use HeaderCheckWorker.class.getSimpleName() to make it refactoring-safer?